### PR TITLE
Add emle-mace backend, MACEEMLEJoint model, and ORCA data utilities

### PR DESCRIPTION
**Pull request by kzinovjev**
## New backend: `emle-mace`

A new `MACEEMLEJoint` model class (`emle/models/_mace.py`) wraps an `EnergyEMLEMACE` model and computes static and induced EMLE energy components directly from its outputs, without requiring a separate EMLE model file.

`EMLECalculator` now supports `backend="emle-mace"` and routes to `MACEEMLEJoint`.

A new `use_dipoles` flag is introduced in `EMLECalculator` (propagated to `emle-server` via `EMLE_USE_DIPOLES`) to enable static atomic dipoles. This feature requires an equivariant model and is currently only supported by `emle-mace` backend.

The `EnergyEMLEMACE` model implementation and training protocol is hosted in a separate repository:  
https://github.com/kzinovjev/emle-mace  
The refactoring required to interface `emle-engine` with `emle-mace` package was performed with Claude Code (Sonnet 4.6) based on interface with previous `EnergyEMLEMACE` implementation directly in a fork of `mace` package.

`emle-mace` is an optional dependency; existing backends remain unaffected if it is not installed.

---

## EnergyEMLEMACE model loading

- `MACEEMLE` and `MACEEMLEJoint` detect `EnergyEMLEMACE` checkpoints via the model class name and dispatch to `extract_config_emle_mace_model` (from `emle-mace`) for config extraction. All other models fall back to the standard MACE extractor.  
- `EnergyEMLEMACE` is injected into `mace.modules.models` at import time, enabling `torch.load()` to deserialize checkpoints without modifying the MACE package.  
- Compiled models (`TorchScript` / `RecursiveScriptModule`) are handled by reusing the loaded module directly, bypassing config extraction (unsupported for compiled models).  
- `torch.load(..., weights_only=False)` is now explicitly used for local files to support PyTorch 2.6+.  
- `torch.serialization.add_safe_globals([slice])` is added to allow `e3nn` (which stores a `slice` in `constants.pt`) to load under PyTorch 2.6+ without requiring `TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD`.

---

## Static atomic dipoles support in EMLE

- `EMLE.forward` now accepts an `external_params` argument, allowing callers to provide precomputed EMLE properties:
  - `s`, `q_core`, `q_val`, `mu`, `k_Z`, `a_Thole`  
  instead of running `EMLEBase`.  

- `EMLEBase.get_static_energy` now accepts an optional `mu` argument. When provided, the dipole contribution (`vpot_mu`) is included in the electrostatic energy.  

This mechanism is used by `MACEEMLEJoint` to incorporate static atomic dipoles predicted by `EnergyEMLEMACE`.

---

## EMLEAnalyzer improvements

- Added support for `emle-mace` backend:
  - EMLE properties (`s`, `q_core`, `q`, `q_val`, `mu`) are read from `backend.emle_values`.  
  - Thole matrix is reconstructed from learned model parameters.  

- MACE gradients (`grad_backend`) are computed via `torch.autograd.grad` and exported.  
- Static dipole energy (`e_static_mu`) is computed and exported when `use_dipoles=True`.  
- `q` and `mu` are now included in exported outputs.  
- Backend calls now pass `qm_charge` explicitly.

---

## ORCA data utilities

- New `ORCAParser` methods:
  - `get_E_vac` — parses vacuum energies  
  - `get_forces` — parses Cartesian gradients (converted to forces in eV/Å)  

  These are automatically invoked during parsing.

- Removed module-local `_HARTREE_TO_KCALMOL`; all unit conversions now use the shared `_units` module.  
- Added `_HARTREE_BOHR_TO_EV_A` constant to `_units.py`.  

- New module: `emle/_tarball_to_extXYZ.py`
  - Provides `orca_to_extxyz()` to convert ORCA tarballs to `extXYZ` format using `ORCAParser` and ASE.  
  - Outputs energies, forces, and EMLE properties as per-atom arrays and metadata fields.  

- New CLI entry point:
  ```bash
  bin/tarball-to-extxyz

### CODE REVIEW

**Review by lohedges — APPROVED** [review]
Many thanks for this addition. I've check locally and everything works as expected. I'm not sure why the CI didn't run. I'll sort out any issues if it doesn't work on merge.

**Comment by lohedges** [comment]
CI ran fine on the merge and (as expected) everything passed. Thanks again.
